python/pyspark/sql/functions.py

```diff
@@ -1424,7 +1424,9 @@ def substring(str, pos, len):
     """
     Substring starts at `pos` and is of length `len` when str is String type or
     returns the slice of byte array that starts at `pos` in byte and is of length `len`
-    when str is Binary type
+    when str is Binary type.
+    
```

Comment:
Ah, PEP8 doens't like the trailing whitespace:
`./python/pyspark/sql/functions.py:1428:1: W293 blank line contains whitespace`